TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -0,0 +1,139 @@
+package org.opensearch.migrations.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.exporter.otlp.logs.OtlpGrpcLogRecordExporter;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.opensearch.migrations.Utils;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+public class RootOtelContext implements IRootOtelContext {
+    private final OpenTelemetry openTelemetryImpl;
+    private final String scopeName;
+    @Getter
+    @Setter
+    Exception observedExceptionToIncludeInMetrics;
+
+    public static OpenTelemetry initializeOpenTelemetryForCollector(@NonNull String collectorEndpoint,
```

Comment:
nit: I prefer limiting public methods to reduce integration choices later (i.e. have other parts of code call null-safe method)

Reply:
Done

Reply:
Why would you want to reduce integration choices later?  The initializeOpenTelemetryWithCollectorOrAsNoop has a looser contract.  If a caller wants to make sure that they create a collector-backed OpenTelemetry object, they would need to call the loose one, then do their own checks.  It seems like it's a noop at the moment for us to maintain both?
I actually see the outer one as a convenience function.  I could see a method to return the empty factory which takes no arguments, which I've also added (meaning that there are now 3 public functions).
I could see keeping these as protected methods, but it's too cumbersome to overload all of the different constructors for each subclass, hence the reason that I'm putting it directly into the caller's hands.  I've also already committed to making Otel classes part of the API, so the decisions fit together, in my head... What do you think?